test/hooks.test.js

```diff
@@ -1269,8 +1269,16 @@ test('clear payload', t => {
 })
 
 test('onSend hook throws', t => {
-  t.plan(9)
-  const fastify = Fastify()
+  t.plan(12)
+  const MockedFastify = t.mock('../', {
+    '../lib/schemas': {
+      getSchemaSerializer: (param1, param2, param3) => {
+        t.equal(param3, 'application/json; charset=utf-8', 'param3 should be "application/json; charset=utf-8"')
+        t.end()
+      }
+    }
+  })
```

Comment:
I didn't know that this needs to be done like this, thank you @climba03003 


I am now trying to figure out why 
```js
 FAIL ​ test/hooks.test.js
 ✖ this.opts.bucket is not a function

    } else {
      this.schemaBucket = this.opts.bucket()
------------------------------------^
    }
  }

  test: onSend hook throws
  stack: |
    new SchemaController (lib/schema-controller.js:55:37)
    Function.buildSchemaController (lib/schema-controller.js:37:10)
    fastify (fastify.js:206:45)
    Test.<anonymous> (test/hooks.test.js:1281:19)
  at:
    line: 55
    column: 37
    file: lib/schema-controller.js
    function: SchemaController
  type: TypeError
  tapCaught: testFunctionThrow
```
is happening.

Reply:
Maybe it would be easier to use `proxyquire`. It can replace the necessary only.

```js
const proxyquire = require('proxyquire')
const Fastify = proxyquire('..', {
  '../lib/schemas': {
    getSchemaSerializer: (param1, param2, param3) => {
      t.equal(param3, 'application/json; charset=utf-8', 'param3 should be "application/json; charset=utf-8"')
      t.end()
    }
  }
})
```